apis/control/v1alpha1/k8ssandratask_types_test.go

```diff
@@ -0,0 +1,93 @@
+package v1alpha1
+
+import (
+	"testing"
+	"time"
+
+	"github.com/k8ssandra/cass-operator/apis/control/v1alpha1"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestRefreshGlobalStatus(t *testing.T) {
+	// setup Task status data
+	dcMap := generateDatacenters()
+
+	kts := K8ssandraTaskStatus{
+		CassandraTaskStatus: v1alpha1.CassandraTaskStatus{},
+		Datacenters:         dcMap,
+	}
+	kt := K8ssandraTask{
+		Status: kts,
+	}
+	kt.RefreshGlobalStatus(3)
+	assert.Equal(t, 2, kt.Status.Active)
+	assert.Equal(t, 1, kt.Status.Failed)
+	assert.Equal(t, 1, kt.Status.Succeeded)
+	assert.Equal(t, 0, kt.Status.StartTime.Compare(time.Date(2023, 6, 10, 10, 10, 10, 10, time.UTC)))
+	assert.Equal(t, v1.ConditionTrue, kt.GetConditionStatus(v1alpha1.JobRunning))
+	assert.Equal(t, v1.ConditionTrue, kt.GetConditionStatus(v1alpha1.JobFailed))
+	// the test data does not have a completion time, so it should be Unknown
+	assert.Equal(t, v1.ConditionUnknown, kt.GetConditionStatus(v1alpha1.JobComplete))
+}
+
+func generateDatacenters() map[string]v1alpha1.CassandraTaskStatus {
+	dcMap := make(map[string]v1alpha1.CassandraTaskStatus, 3)
```

Comment:
This can be written as a map literal:
```go
return map[string]v1alpha1.CassandraTaskStatus {
  "dc1": v1alpha1.CassandraTaskStatus{ ... },
  "dc2": v1alpha1.CassandraTaskStatus{ ... },
  "dc3": v1alpha1.CassandraTaskStatus{ ... },
}
```